pkg/validation/components.go

```diff
@@ -60,6 +60,41 @@ func ValidateComponents(components []v1alpha2.Component) (returnedErr error) {
 					returnedErr = multierror.Append(returnedErr, reservedEnvErr)
 				}
 			}
+			var memoryLimit, cpuLimit, memoryRequest, cpuRequest resource.Quantity
+			if component.Container.MemoryLimit != "" {
+				memoryLimit, err = resource.ParseQuantity(component.Container.MemoryLimit)
+				if err != nil {
+					parseQuantityErr := &ParsingResourceRequirementError{resource: MemoryLimit, cmpName: component.Name, errMsg: err.Error()}
+					returnedErr = multierror.Append(returnedErr, parseQuantityErr)
+				}
+			}
+			if component.Container.CpuLimit != "" {
+				cpuLimit, err = resource.ParseQuantity(component.Container.CpuLimit)
+				if err != nil {
+					parseQuantityErr := &ParsingResourceRequirementError{resource: CpuLimit, cmpName: component.Name, errMsg: err.Error()}
+					returnedErr = multierror.Append(returnedErr, parseQuantityErr)
+				}
+			}
+			if component.Container.MemoryRequest != "" {
+				memoryRequest, err = resource.ParseQuantity(component.Container.MemoryRequest)
+				if err != nil {
+					parseQuantityErr := &ParsingResourceRequirementError{resource: MemoryRequest, cmpName: component.Name, errMsg: err.Error()}
+					returnedErr = multierror.Append(returnedErr, parseQuantityErr)
+				} else if !memoryLimit.Equal(resource.Quantity{}) && memoryRequest.Cmp(memoryLimit) > 0 {
+					invalidResourceRequest := &InvalidResourceRequestError{cmpName: component.Name, errMsg: fmt.Sprintf("memoryRequest is greater than memoryLimit.")}
+					returnedErr = multierror.Append(returnedErr, invalidResourceRequest)
+				}
+			}
+			if component.Container.CpuRequest != "" {
+				cpuRequest, err = resource.ParseQuantity(component.Container.CpuRequest)
+				if err != nil {
+					parseQuantityErr := &ParsingResourceRequirementError{resource: CpuRequest, cmpName: component.Name, errMsg: err.Error()}
+					returnedErr = multierror.Append(returnedErr, parseQuantityErr)
+				} else if !cpuLimit.Equal(resource.Quantity{}) && cpuRequest.Cmp(cpuLimit) > 0 {
```

Comment:
```suggestion
				} else if !cpuLimit.IsZero() && cpuRequest.Cmp(cpuLimit) > 0 {
```
wondering if this would be ideal

Reply:
updated